dvc/output/http.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import unicode_literals
+
+from dvc.path import Path
+from dvc.utils.compat import urlparse, urljoin
+from dvc.remote.http import RemoteHTTP
+
+from . import Output
+
+
+class OutputHTTP(Output):
+    REMOTE = RemoteHTTP
+
+    def __init__(self, *args, **kwargs):
+        super(OutputHTTP, self).__init__(*args, **kwargs)
+        assert self.dependency
```

Comment:
btw, why are we using asserts a lot? they should be disabled if we use optimized mode with Python. Which we should be using probably for production packages. There is a possibility that other libraries code would benefit from it.

Reply:
To ensure that we don't mess up. I didn't actually know about the optimized mode. Indeed, we could build our binary packages with it. Created https://github.com/iterative/dvc/issues/2088